File: sdi3-23.Cli-EJB/src/uo/sdi/client/util/Printer.java
package uo.sdi.client.util;

import java.util.List;

import uo.sdi.dto.UserInfoDTO;
import alb.util.console.Console;

public class Printer {

    private static final String TAB = "    ";

    public static void printUsers(List<UserInfoDTO> users) {
	Console.println("Listado de usuarios:\n");

	for (UserInfoDTO user : users) {
	    String mensaje = TAB + "id: %d - login: %s - email: %s - "
		    + "estado: %s - admin: %s\n";

	    String isAdmin;

	    if (user.getIsAdmin()) {
		isAdmin = "si";
	    }

	    else {
		isAdmin = "no";
	    }

	    mensaje += TAB + TAB;
	    mensaje += "--> Número de tareas planeadas (sin finalizar): %d\n";

	    mensaje += TAB + TAB;
	    mensaje += "--> Número de tareas no planeadas (sin finalizar): %d\n";

	    mensaje += TAB + TAB;
	    mensaje += "--> Número de tareas retrasadas y finalizadas: %d\n";

	    mensaje += TAB + TAB;
	    mensaje += "--> Número de tareas finalizadas: %d\n\n";

	    Console.printf(mensaje, user.getId(), user.getLogin(),
		    user.getEmail(), user.getStatus().name().toLowerCase(),
		    isAdmin, user.getNumPlannedTasks(),
		    user.getNumNotPlannedTasks(),
		    user.getNumFinishedDelayedTasks(),
		    user.getNumFinishedTasks());
	}

	Console.println("--- Fin del listado ---");
    }

}

File: sdi3-23.Cli-SOAPcsharp/sdi3-23.Cli-SOAPcsharp/Printer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace sdi3_23.Cli_SOAPcsharp
{
    class Printer
    {
        public static void printUsers(IList<ServiceSoap.userInfoDTO> users)
        {
            Consola.println("Listado de usuarios:\n");

            foreach (ServiceSoap.userInfoDTO user in users)
            {
                String mensaje = "\tid: %s - login: %s - email: %s - "
            + "estado: %s - admin: %s\n";

                String isAdmin;
                if (user.is_admin)
                {
                    isAdmin = "si";
                }
                else
                {
                    isAdmin = "no";
                }


                mensaje += "\t\t--> Número de tareas planeadas (sin finalizar): %s\n";


                mensaje += "\t\t--> Número de tareas no planeadas (sin finalizar): %s\n";


                mensaje += "\t\t--> Número de tareas retrasadas y finalizadas: %s\n";


                mensaje += "\t\t--> Número de tareas finalizadas: %s\n\n";

                Consola.printf(mensaje, user.id.ToString(), user.login.ToString(), user.email.ToString(),
                    user.status.ToString().ToLower()
                    , isAdmin, user.num_planned_tasks.ToString(), user.num_not_planned_tasks.ToString()
                    , user.num_finished_delayed_tasks.ToString(), user.num_finished_tasks.ToString());

            }
            Consola.println("--- Fin del listado ---");

        }

    }
}
